CMakeLists.txt

```diff
@@ -1,6 +1,13 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 3.1)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.15)
```

Comment:
I think this won't work on Ubuntu 18.04. Unfortunately, there are many robots still living in 18.04.

Do we need to upgrade cmake?

Reply:
I didn't realise this was such a big jump. I've changed the minimum required CMake to 3.3. How is this?